api/group.py

```diff
@@ -184,13 +197,30 @@ def patch_group_by_id(group_id, body, rbac_filter=None):
         abort(HTTPStatus.NOT_FOUND)
 
     try:
-        if get_flag_value(FLAG_INVENTORY_KESSEL_WORKSPACE_MIGRATION):
-            # Update group on Kessel
-            new_group_name = validated_patch_group_data.get("name")
-
-            if new_group_name:
+        # Check if trying to modify ungrouped group before any RBAC calls
+        if group_to_update.ungrouped and "name" in validated_patch_group_data:
+            log_patch_group_failed(logger, group_id)
+            abort(HTTPStatus.BAD_REQUEST, "The 'ungrouped' group can not be modified.")
+
+        if not get_flag_value(FLAG_INVENTORY_KESSEL_PHASE_1) and (
+            new_group_name := validated_patch_group_data.get("name")
+        ):
+            existing_groups = get_group_by_name_from_db(new_group_name, identity.org_id)
+
+            # Check if there are any existing groups with this name other than the current group
+            if len(existing_groups) > 0 and any(group.id != group_to_update.id for group in existing_groups):
```

Comment:
I think it would be simpler to first check if the request is changing the group name, and then just checking whether any groups exist with that name. Then you also don't need `get_group_by_name_from_db`
```suggestion
            if new_group_name != group_to_update.name and does_group_with_name_exist(group_name, get_current_identity().org_id):
```